lib/portage/dbapi/porttree.py

```diff
@@ -637,7 +638,9 @@ def _pull_valid_cache(self, cpv, ebuild_path, repo_path):
                 # EAPI from _parse_eapi_ebuild_head, we disregard cache entries
                 # for unsupported EAPIs.
                 continue
-            if auxdb.validate_entry(metadata, ebuild_hash, eclass_db):
+            if not repo.volatile or auxdb.validate_entry(
+                metadata, ebuild_hash, eclass_db
+            ):
```

Comment:
I feel like we probably need a _pull_valid_cache to indicate cases like egencache where the entries are expected to be invalid.

Reply:
Ah, good point. But this is enough to make me think we should revisit this separately from the git changes in order to get those in. Thank you!